fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/ChunkHolderMixin.java

```
@@ -0,0 +1,140 @@
+	 */
+	@Inject(method = "updateFutures", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/world/ChunkHolder;combineSavingFuture(Ljava/util/concurrent/CompletableFuture;)V", shift = At.Shift.AFTER, ordinal = 0))
+	private void updateFutures$inaccessibleToFull(ServerChunkLoadingManager chunkLoadingManager, Executor executor, CallbackInfo ci) {
+		if (this.getUncheckedOrNull(ChunkStatus.FULL) instanceof WorldChunk && this.currentEventLevelType == INACCESSIBLE) { // prevent duplicate events with ChunkGeneratingMixin
+			ServerChunkEvents.CHUNK_LEVEL_TYPE_CHANGE.invoker().onChunkLevelTypeChange((ServerWorld) world, this.pos, INACCESSIBLE, FULL);
+			this.currentEventLevelType = FULL;
+		}
+	}
+
+	/**
+	 * Handles FULL -> BLOCK_TICKING.
+	 */
+	@Inject(method = "updateFutures", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/world/ChunkHolder;combineSavingFuture(Ljava/util/concurrent/CompletableFuture;)V", shift = At.Shift.AFTER, ordinal = 1))
+	private void updateFutures$fullToBlockTicking(ServerChunkLoadingManager chunkLoadingManager, Executor executor, CallbackInfo ci) {
+		this.fullToBlockTickingEvent = this.tickingFuture.thenAccept(optionalChunk -> {
+			optionalChunk.ifPresent(worldChunk -> {
+				ServerChunkEvents.CHUNK_LEVEL_TYPE_CHANGE.invoker().onChunkLevelTypeChange((ServerWorld) world, this.pos, FULL, BLOCK_TICKING);
+				this.currentEventLevelType = BLOCK_TICKING;
+			});
+		});
+	}
+
+	/**
+	 * Handles FULL -> BLOCK_TICKING.
+	 */
+	@Inject(method = "updateFutures", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/world/ChunkHolder;combineSavingFuture(Ljava/util/concurrent/CompletableFuture;)V", shift = At.Shift.AFTER, ordinal = 2))
+	private void updateFutures$blockTickingToEntityTicking(ServerChunkLoadingManager chunkLoadingManager, Executor executor, CallbackInfo ci) {
+		this.fullToBlockTickingEvent.thenAccept(optionalChunk -> {
```

Same exception swallowing issue, see comment above.